src/include/OSL/rendererservices.h

```diff
@@ -29,6 +30,125 @@ typedef const void* TransformationPtr;
 typedef void (*PrepareClosureFunc)(RendererServices*, int id, void* data);
 typedef void (*SetupClosureFunc)(RendererServices*, int id, void* data);
 
+enum class AttributeSpecBuiltinArg {
+    ShaderGlobalsPointer,  // void* (ideally ShaderGlobals*)
+    ShadeIndex,            // int
+    Derivatives,           // int (ideally bool)
+    Type,                  // long long (ideally TypeDesc)
+    ArrayIndex,            // int, Always zero for non-indexed array lookups.
+    ArrayLookup,           // int (ideally bool)
+    ObjectName,            // const char* (ideally ustring)
```

Comment:
Any reason the `const char* (ideally ustring)` can't be explicitly ustring?  IE   `const char* (ustring)`
(I realize the parameter itself is still const char * as it crosses the llvm JIT and llvm precompiled C++ boundary.)